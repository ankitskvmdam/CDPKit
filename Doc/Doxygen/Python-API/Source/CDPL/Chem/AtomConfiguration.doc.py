#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) Thomas Seidel <thomas.seidel@univie.ac.at>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program; see the file COPYING. If not, write to
# the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
# Boston, MA 02111-1307, USA.
#

##
# \brief Provides constants that are used to specify the <em>R/S</em> configuration of atoms.
# 
class AtomConfiguration(Boost.Python.instance):

    ##
    # \brief Specifies that the configuration of the atom is undefined.
    # 
    UNDEF = 0

    ##
    # \brief Specifies that the atom is not a stereogenic center and thus cannot be assigned a configuration.
    # 
    NONE = 1

    ##
    # \brief Specifies that the atom has <em>R</em> configuration.
    # 
    R = 2

    ##
    # \brief Specifies that the atom has <em>S</em> configuration.
    # 
    S = 4

    ##
    # \brief Specifies that the atom is a stereogenic center but has no defined configuration.
    # 
    EITHER = 8

    ##
    # \brief Specifies that the atom has <em>r</em> configuration.
    # 
    r = 16

    ##
    # \brief Specifies that the atom has <em>s</em> configuration.
    # 
    s = 17

    ##
    # \brief Specifies that the atom has <em>M</em> configuration.
    # 
    M = 19

    ##
    # \brief Specifies that the atom has <em>P</em> configuration.
    # 
    P = 18

    ##
    # \brief Specifies that the atom has <em>m</em> configuration.
    # 
    m = 21

    ##
    # \brief Specifies that the atom has <em>p</em> configuration.
    # 
    p = 20

    ##
    # \brief Specifies that the geometry of the ligand arrangement is <em>square planar</em>.
    # 
    SP = 22

    ##
    # \brief Specifies that the geometry of the ligand arrangement is <em>trigonal bipyramidal</em>.
    # 
    TB = 26

    ##
    # \brief Specifies that the geometry of the ligand arrangement is <em>octahedral</em>.
    # 
    OH = 47
