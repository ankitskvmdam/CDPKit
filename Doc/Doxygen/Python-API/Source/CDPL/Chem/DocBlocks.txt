#
# --- Unresolved property getter/setter functions ---
#
