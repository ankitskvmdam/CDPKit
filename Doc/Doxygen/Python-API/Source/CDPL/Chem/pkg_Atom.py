#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) Thomas A. Seidel <thomas.seidel@univie.ac.at>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program; see the file COPYING. If not, write to
# the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
# Boston, MA 02111-1307, USA.
#

##
# \brief 
#
class Atom(AtomContainer, BondContainer, Entity3D):

    ##
    # \brief 
    #
    class AtomSequence(Boost.Python.instance):

        ##
        # \brief 
        # \return 
        #
        def __len__() -> int: pass

        ##
        # \brief 
        # \param idx 
        # \return 
        #
        def __getitem__(idx: int) -> Atom: pass

        ##
        # \brief Returns the result of the membership test operation <tt>atom in self</tt>.
        # \param atom The value to test for membership.
        # \return The result of the membership test operation.
        #
        def __contains__(atom: Atom) -> bool: pass

    ##
    # \brief 
    #
    class BondSequence(Boost.Python.instance):

        ##
        # \brief 
        # \return 
        #
        def __len__() -> int: pass

        ##
        # \brief 
        # \param idx 
        # \return 
        #
        def __getitem__(idx: int) -> Bond: pass

        ##
        # \brief Returns the result of the membership test operation <tt>bond in self</tt>.
        # \param bond The value to test for membership.
        # \return The result of the membership test operation.
        #
        def __contains__(bond: Bond) -> bool: pass

    ##
    # \brief Initializes the \e %Atom instance.
    #
    def __init__() -> None: pass

    ##
    # \brief 
    # \return 
    #
    def getMolecule() -> Molecule: pass

    ##
    # \brief 
    # \param atom 
    # \return 
    #
    def getBondToAtom(atom: Atom) -> Bond: pass

    ##
    # \brief 
    # \param atom 
    # \return 
    #
    def findBondToAtom(atom: Atom) -> Bond: pass

    ##
    # \brief 
    # \return 
    #
    def getIndex() -> int: pass

    ##
    # \brief Replaces the current state of \a self with a copy of the state of the \e %Atom instance \a atom.
    # \param atom The \e %Atom instance to copy.
    # \return The assignment target \a self.
    #
    def assign(atom: Atom) -> Atom: pass

    ##
    # \brief 
    # \return 
    #
    def getAtoms() -> object: pass

    ##
    # \brief 
    # \return 
    #
    def getBonds() -> object: pass

    ##
    # \brief 
    # \param idx 
    # \return 
    #
    def getAtom(idx: int) -> Atom: pass

    ##
    # \brief 
    # \param atom 
    # \return 
    #
    def containsAtom(atom: Atom) -> bool: pass

    ##
    # \brief 
    # \param arg1 
    # \param arg2 
    #
    def containsAtom(arg2: Atom) -> None: pass

    ##
    # \brief 
    # \param atom 
    # \return 
    #
    def getAtomIndex(atom: Atom) -> int: pass

    ##
    # \brief 
    # \param arg1 
    # \param arg2 
    #
    def getAtomIndex(arg2: Atom) -> None: pass

    ##
    # \brief 
    # \return 
    #
    def getNumAtoms() -> int: pass

    ##
    # \brief 
    # \param func 
    #
    def orderAtoms(func: CDPL.ForceField.InteractionFilterFunction2) -> None: pass

    ##
    # \brief 
    # \param idx 
    # \return 
    #
    def getEntity(idx: int) -> Entity3D: pass

    ##
    # \brief 
    # \return 
    #
    def getNumEntities() -> int: pass

    ##
    # \brief 
    # \param idx 
    # \return 
    #
    def getBond(idx: int) -> Bond: pass

    ##
    # \brief 
    # \param bond 
    # \return 
    #
    def containsBond(bond: Bond) -> bool: pass

    ##
    # \brief 
    # \param arg1 
    # \param arg2 
    #
    def containsBond(arg2: Bond) -> None: pass

    ##
    # \brief 
    # \param func 
    #
    def orderBonds(func: BoolBond2Functor) -> None: pass

    ##
    # \brief 
    # \param bond 
    # \return 
    #
    def getBondIndex(bond: Bond) -> int: pass

    ##
    # \brief 
    # \param arg1 
    # \param arg2 
    #
    def getBondIndex(arg2: Bond) -> None: pass

    ##
    # \brief 
    # \return 
    #
    def getNumBonds() -> int: pass

    ##
    # \brief 
    # \param key 
    # \param value 
    #
    def setProperty(key: CDPL.Base.LookupKey, value: CDPL.Base.Variant) -> None: pass

    ##
    # \brief 
    # \param key 
    # \return 
    #
    def removeProperty(key: CDPL.Base.LookupKey) -> bool: pass

    ##
    # \brief 
    # \param key 
    # \param throw_ 
    # \return 
    #
    def getProperty(key: CDPL.Base.LookupKey, throw_: bool = False) -> CDPL.Base.Variant: pass

    ##
    # \brief 
    # \param key 
    # \return 
    #
    def isPropertySet(key: CDPL.Base.LookupKey) -> bool: pass

    ##
    # \brief 
    #
    def clearProperties() -> None: pass

    ##
    # \brief 
    # \param cntnr 
    #
    def addProperties(cntnr: CDPL.Base.PropertyContainer) -> None: pass

    ##
    # \brief 
    # \param cntnr 
    #
    def copyProperties(cntnr: CDPL.Base.PropertyContainer) -> None: pass

    ##
    # \brief 
    # \param cntnr 
    #
    def swap(cntnr: CDPL.Base.PropertyContainer) -> None: pass

    ##
    # \brief Returns the result of the membership test operation <tt>atom in self</tt>.
    # \param atom The value to test for membership.
    # \return The result of the membership test operation.
    #
    def __contains__(atom: Atom) -> bool: pass

    ##
    # \brief Returns the result of the membership test operation <tt>bond in self</tt>.
    # \param bond The value to test for membership.
    # \return The result of the membership test operation.
    #
    def __contains__(bond: Bond) -> bool: pass

    ##
    # \brief Returns the result of the membership test operation <tt>key in self</tt>.
    # \param key The value to test for membership.
    # \return The result of the membership test operation.
    #
    def __contains__(key: CDPL.Base.LookupKey) -> bool: pass

    ##
    # \brief 
    # \param key 
    # \return 
    #
    def __getitem__(key: CDPL.Base.LookupKey) -> CDPL.Base.Variant: pass

    ##
    # \brief 
    # \param key 
    # \param value 
    #
    def __setitem__(key: CDPL.Base.LookupKey, value: CDPL.Base.Variant) -> None: pass

    ##
    # \brief 
    # \param key 
    # \return 
    #
    def __delitem__(key: CDPL.Base.LookupKey) -> bool: pass

    ##
    # \brief 
    # \return 
    #
    def __len__() -> int: pass

    ##
    # \brief 
    #
    molecule = property(getMolecule)

    ##
    # \brief 
    #
    index = property(getIndex)

    ##
    # \brief 
    #
    atoms = property(getAtoms)

    ##
    # \brief 
    #
    bonds = property(getBonds)
