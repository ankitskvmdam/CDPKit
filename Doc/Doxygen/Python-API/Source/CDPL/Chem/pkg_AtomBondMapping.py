#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) Thomas A. Seidel <thomas.seidel@univie.ac.at>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program; see the file COPYING. If not, write to
# the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
# Boston, MA 02111-1307, USA.
#

##
# \brief 
#
class AtomBondMapping(Boost.Python.instance):

    ##
    # \brief Initializes the \e %AtomBondMapping instance.
    #
    def __init__() -> None: pass

    ##
    # \brief Initializes the \e %AtomBondMapping instance.
    # \param mapping 
    #
    def __init__(mapping: AtomBondMapping) -> None: pass

    ##
    # \brief Returns the numeric identifier (ID) of the wrapped C++ class instance.
    #
    # Different Python \e %AtomBondMapping instances may reference the same underlying C++ class instance. The commonly used Python expression
    # <tt>a is not b</tt> thus cannot tell reliably whether the two \e %AtomBondMapping instances \e a and \e b reference different C++ objects. 
    # The numeric identifier returned by this method allows to correctly implement such an identity test via the simple expression
    # <tt>a.getObjectID() != b.getObjectID()</tt>.
    #
    # \return The numeric ID of the internally referenced C++ class instance.
    #
    def getObjectID() -> int: pass

    ##
    # \brief Replaces the current state of \a self with a copy of the state of the \e %AtomBondMapping instance \a mapping.
    # \param mapping The \e %AtomBondMapping instance to copy.
    # \return The assignment target \a self.
    #
    def assign(mapping: AtomBondMapping) -> AtomBondMapping: pass

    ##
    # \brief 
    # \return 
    #
    def getAtomMapping() -> AtomMapping: pass

    ##
    # \brief 
    # \return 
    #
    def getBondMapping() -> BondMapping: pass

    ##
    # \brief 
    #
    def clear() -> None: pass

    ##
    # \brief Returns the result of the comparison operation <tt>self == mapping</tt>.
    # \param mapping The \e %AtomBondMapping instance to be compared with.
    # \return The result of the comparison operation.
    #
    def __eq__(mapping: AtomBondMapping) -> bool: pass

    ##
    # \brief Returns the result of the comparison operation <tt>self != mapping</tt>.
    # \param mapping The \e %AtomBondMapping instance to be compared with.
    # \return The result of the comparison operation.
    #
    def __ne__(mapping: AtomBondMapping) -> bool: pass

    ##
    # \brief 
    #
    objectID = property(getObjectID)

    ##
    # \brief 
    #
    atomMapping = property(getAtomMapping)

    ##
    # \brief 
    #
    bondMapping = property(getBondMapping)
