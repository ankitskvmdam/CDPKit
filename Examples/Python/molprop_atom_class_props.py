#!/bin/env python

##
# molprop_atom_class_props.py 
#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) 2003-2022 Thomas A. Seidel <thomas.seidel@univie.ac.at>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program; see the file COPYING. If not, write to
# the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
# Boston, MA 02111-1307, USA.
##


import sys
import os

import CDPL.Chem as Chem
import CDPL.MolProp as MolProp


# function called for read molecule
def procMolecule(mol: Chem.MolecularGraph) -> None:
    Chem.calcImplicitHydrogenCounts(mol, False)  # calculate implicit hydrogen counts and set corresponding property for all atoms
    Chem.perceiveHybridizationStates(mol, False) # perceive atom hybridization and set corresponding property for all atoms
    Chem.perceiveSSSR(mol, False)                # perceive smallest set of smallest rings and store as Chem.MolecularGraph property
    Chem.setRingFlags(mol, False)                # perceive cycles and set corresponding atom and bond properties
    Chem.setAromaticityFlags(mol, False)         # perceive aromaticity and set corresponding atom and bond properties

    for atom in mol.atoms:
        print('- Atom #%s' % str(atom.getIndex()))
        print('\tIs std. hydrogen: %s' % str(MolProp.isOrdinaryHydrogen(atom, mol)))
        print('\tIs heavy atom: %s' % str(MolProp.isHeavy(atom)))
        print('\tIs unsaturated: %s' % str(MolProp.isUnsaturated(atom, mol)))
        print('\tIs hydrogen acceptor: %s' % str(MolProp.isHydrogenAcceptor(atom, mol)))
        print('\tIs hydrogen donor: %s' % str(MolProp.isHydrogenDonor(atom, mol)))
        print('\tIs carbonyl carbon: %s' % str(MolProp.isCarbonylLikeAtom(atom, mol, True, True)))
        print('\tIs amide carbon: %s' % str(MolProp.isAmideCenterAtom(atom, mol, True, True)))
        print('\tIs amide nitrogen: %s' % str(MolProp.isAmideNitrogen(atom, mol, True, True)))
        print('\tIs invertible nitrogen: %s' % str(MolProp.isInvertibleNitrogen(atom, mol)))
        print('\tIs planar nitrogen: %s' % str(MolProp.isPlanarNitrogen(atom, mol)))
        
def getReaderByFileExt(filename: str) -> Chem.MoleculeReader:
    # get the extension of the input file
    name_and_ext = os.path.splitext(filename)

    if name_and_ext[1] == '':
        sys.exit('Error: could not determine input file format (file extension missing).')

    # get input handler for the format specified by the input file's extension
    ipt_handler = Chem.MoleculeIOManager.getInputHandlerByFileExtension(name_and_ext[1][1:].lower())

    if not ipt_handler:
        sys.exit('Error: unknown input file format \'%s\'' % name_and_ext[1])

    # return file reader instance
    return ipt_handler.createReader(filename)
    
def main() -> None:
    if len(sys.argv) < 2:
        sys.exit('Usage: %s <input file>' % sys.argv[0])

    # if input are expected to be in a specific format, a reader for the specific format could be create directly,
    # e.g. reader = Chem.FileSDFMoleculeReader(sys.argv[1])
    reader = getReaderByFileExt(sys.argv[1]) 
    
    # create an instance of the default implementation of the Chem.Molecule interface
    mol = Chem.BasicMolecule()
    
    # read and process molecules one after the other until the end of input has been reached
    try:
        if reader.read(mol):
            try:
                procMolecule(mol)
            except Exception as e:
                sys.exit('Error: processing of molecule failed:\n' + str(e))
                
    except Exception as e: # handle exception raised in case of severe read errors
        sys.exit('Error: reading molecule failed:\n' + str(e))

    sys.exit(0)
        
if __name__ == '__main__':
    main()

