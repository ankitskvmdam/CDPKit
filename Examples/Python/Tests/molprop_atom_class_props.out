- Atom #0
	Is std. hydrogen: False
	Is heavy atom: True
	Is unsaturated: False
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #1
	Is std. hydrogen: False
	Is heavy atom: True
	Is unsaturated: True
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #2
	Is std. hydrogen: False
	Is heavy atom: True
	Is unsaturated: True
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #3
	Is std. hydrogen: False
	Is heavy atom: True
	Is unsaturated: True
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #4
	Is std. hydrogen: False
	Is heavy atom: True
	Is unsaturated: True
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #5
	Is std. hydrogen: False
	Is heavy atom: True
	Is unsaturated: False
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #6
	Is std. hydrogen: False
	Is heavy atom: True
	Is unsaturated: False
	Is hydrogen acceptor: True
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #7
	Is std. hydrogen: False
	Is heavy atom: True
	Is unsaturated: False
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #8
	Is std. hydrogen: False
	Is heavy atom: True
	Is unsaturated: True
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #9
	Is std. hydrogen: False
	Is heavy atom: True
	Is unsaturated: True
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #10
	Is std. hydrogen: False
	Is heavy atom: True
	Is unsaturated: True
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #11
	Is std. hydrogen: False
	Is heavy atom: True
	Is unsaturated: True
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #12
	Is std. hydrogen: False
	Is heavy atom: True
	Is unsaturated: True
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #13
	Is std. hydrogen: False
	Is heavy atom: True
	Is unsaturated: True
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #14
	Is std. hydrogen: True
	Is heavy atom: False
	Is unsaturated: False
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #15
	Is std. hydrogen: False
	Is heavy atom: True
	Is unsaturated: True
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #16
	Is std. hydrogen: False
	Is heavy atom: True
	Is unsaturated: True
	Is hydrogen acceptor: True
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #17
	Is std. hydrogen: True
	Is heavy atom: False
	Is unsaturated: False
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #18
	Is std. hydrogen: True
	Is heavy atom: False
	Is unsaturated: False
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #19
	Is std. hydrogen: True
	Is heavy atom: False
	Is unsaturated: False
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #20
	Is std. hydrogen: True
	Is heavy atom: False
	Is unsaturated: False
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #21
	Is std. hydrogen: False
	Is heavy atom: True
	Is unsaturated: False
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #22
	Is std. hydrogen: False
	Is heavy atom: True
	Is unsaturated: False
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #23
	Is std. hydrogen: False
	Is heavy atom: True
	Is unsaturated: False
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #24
	Is std. hydrogen: False
	Is heavy atom: True
	Is unsaturated: False
	Is hydrogen acceptor: True
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: True
	Is planar nitrogen: False
- Atom #25
	Is std. hydrogen: False
	Is heavy atom: True
	Is unsaturated: False
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #26
	Is std. hydrogen: True
	Is heavy atom: False
	Is unsaturated: False
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #27
	Is std. hydrogen: True
	Is heavy atom: False
	Is unsaturated: False
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #28
	Is std. hydrogen: True
	Is heavy atom: False
	Is unsaturated: False
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #29
	Is std. hydrogen: False
	Is heavy atom: True
	Is unsaturated: False
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #30
	Is std. hydrogen: True
	Is heavy atom: False
	Is unsaturated: False
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #31
	Is std. hydrogen: True
	Is heavy atom: False
	Is unsaturated: False
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #32
	Is std. hydrogen: True
	Is heavy atom: False
	Is unsaturated: False
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #33
	Is std. hydrogen: True
	Is heavy atom: False
	Is unsaturated: False
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #34
	Is std. hydrogen: True
	Is heavy atom: False
	Is unsaturated: False
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #35
	Is std. hydrogen: True
	Is heavy atom: False
	Is unsaturated: False
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #36
	Is std. hydrogen: True
	Is heavy atom: False
	Is unsaturated: False
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #37
	Is std. hydrogen: True
	Is heavy atom: False
	Is unsaturated: False
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #38
	Is std. hydrogen: True
	Is heavy atom: False
	Is unsaturated: False
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #39
	Is std. hydrogen: False
	Is heavy atom: True
	Is unsaturated: True
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #40
	Is std. hydrogen: False
	Is heavy atom: True
	Is unsaturated: True
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #41
	Is std. hydrogen: True
	Is heavy atom: False
	Is unsaturated: False
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #42
	Is std. hydrogen: True
	Is heavy atom: False
	Is unsaturated: False
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #43
	Is std. hydrogen: True
	Is heavy atom: False
	Is unsaturated: False
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
- Atom #44
	Is std. hydrogen: True
	Is heavy atom: False
	Is unsaturated: False
	Is hydrogen acceptor: False
	Is hydrogen donor: False
	Is carbonyl carbon: False
	Is amide carbon: False
	Is amide nitrogen: False
	Is invertible nitrogen: False
	Is planar nitrogen: False
